src/main/java/core/basesyntax/strategy/DataFileHandler.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.strategy.impl.CsvDataParserImpl;
+import core.basesyntax.strategy.impl.CsvReportBuilderImpl;
+import core.basesyntax.strategy.impl.JsonDataParserImpl;
+import core.basesyntax.strategy.impl.JsonReportBuilderImpl;
+import core.basesyntax.strategy.impl.XmlDataParserImpl;
+import core.basesyntax.strategy.impl.XmlReportBuilderImpl;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public class DataFileHandler {
```

Comment:
it's better to split this class on two, one for forming of parser and the second one for reportBuilders